### PR TITLE
withNewGoogleBucket and withNewCluster / createNewCluster updates

### PR DESCRIPTION
**Pull request by jmthibault79**
Some useful Leo automation test updates I'm using in a forthcoming PR.  Sharing now by request so other users can use them too.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoModelCopy.scala [hunk 1]
```diff
@@ -6,22 +6,15 @@ import java.util.UUID
...
+import org.broadinstitute.dsde.workbench.model.google.{GcsPath, GoogleProject}
 
 sealed trait StringValueClass extends Any
 case class ClusterName(string: String) extends AnyVal with StringValueClass
 case class GoogleServiceAccount(string: String) extends AnyVal with StringValueClass
 case class IP(string: String) extends AnyVal with StringValueClass
 case class OperationName(string: String) extends AnyVal with StringValueClass
 
-/** A GCS relative path */
-case class GcsRelativePath(name: String) extends AnyVal
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoModelCopy.scala]: How come these don't need `with StringValueClass` like the ones above?

Comment (ansingh7115) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoModelCopy.scala]: Oh wait - you're deleting them - sorry I'm being stupid, ignore

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala [hunk 2]
```diff
@@ -272,4 +274,18 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
     testResult.get
   }
 
+  def withNewGoogleBucket[T](googleProject: GoogleProject)(testCode: GcsBucketName => T): T = {
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: So this creates a bucket in a google project using the QA service account. I wonder if this method should take a list of emails to grant (read? owner?) access to. For example if a test needs to access the bucket with Ron, would it work?

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: Not a bad idea.  I do need to grant Reader to Ron('s proxy) in my test.

Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: Should we have the option here to specify a name, like cluster create does?

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: yep

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala [hunk 3]
```diff
@@ -153,9 +154,10 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
 
   def randomClusterName: ClusterName = ClusterName(s"automation-test-a${makeRandomId().toLowerCase}z")
 
-  def createNewCluster(googleProject: GoogleProject, clusterName: ClusterName = randomClusterName)(implicit token: AuthToken): Cluster = {
-    val request = ClusterRequest(Map("foo" -> makeRandomId()))
-    val cluster = createAndMonitor(googleProject, clusterName, request)
+  def defaultClusterRequest: ClusterRequest = ClusterRequest(Map("foo" -> makeRandomId()))
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: Don't miss this unrelated change.  (Snuck in from my other branch but I'm keeping it)

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala (hunk 2 above)
Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: I've been thinking about this some more.  I'm not confident that we'll be able to satisfy all testers' intentions here generically in a simple way.  To do this right, we'd have to consider Bucket access, Object access, and Default Object access as well.  At that point I'm thinking that the interface would have to be something like a collection of (AccessType, AccessLevel, GcsEntity) tuples ... which might be useful but it's starting to smell like a yak.

Comment (rtitle) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: OK, makes sense. Can keep it simple for now.

File: automation/docker/application.conf.ctmpl [hunk 4]
```diff
@@ -27,6 +27,9 @@ fireCloud {
   thurloeApiUrl = "https://thurloe-fiab.dsde-{{$environment}}.broadinstitute.org:25443/"
   {{end}}
   fireCloudId = "{{$secrets.Data.firecloud_id}}"
+
+  # TODO more Config class nonsense
```
Comment (jmthibault79) [inline automation/docker/application.conf.ctmpl]: We're still iterating on automation-test configs in workbench-libs.  This will be part of https://broadinstitute.atlassian.net/browse/GAWB-3110

File: automation/docker/application.conf.ctmpl [hunk 5]
```diff
@@ -62,11 +65,12 @@ users {
   notSoSecretPassword = "{{$users.Data.automation_users_passwd}}"
   userDataPath = "{{$dir}}/src/test/resources/users.json"
 
-  # TODO: these settings only apply to UI but are required by workbench-libs
+  # TODO: these settings only apply to UI but are required by workbench-libs until we fix the org.broadinstitute.dsde.workbench.config.Config class
```
Comment (jmthibault79) [inline automation/docker/application.conf.ctmpl]: ditto the above
